kuma/static/js/wiki-compat-signal.js

```diff
@@ -2,6 +2,669 @@
     'use strict';
 
     var bcTable;
+    var bcSignalBlock;
+    var bcSignalCompleteBlock;
+    var sendReportButton;
+    var nextStepButton;
+    var stepsInfoSpan;
+    var bcSignalStep;
+    var bcSignalSteps = 2;
+
+    /**
+     * Toggles main block visibility and resets form completion progress
+     */
+    function toggleBcSignalBlock () {
+        if (bcSignalCompleteBlock.classList.contains('open')) {
+            bcSignalCompleteBlock.classList.remove('open');
+        } else {
+            bcSignalBlock.classList.toggle('open');
+        }
+
+        resetControls();
+        toStep(1);
+    }
+
+    /**
+     * Moves user to a step, by settings displaying related blocks
+     * @param {Number} step - Step index to be shown
+     */
+    function toStep (step) {
+        if (step !== bcSignalStep) {
+            const prevStep = bcSignalStep;
+            bcSignalStep = step;
+            stepsInfoSpan.innerText = `Step ${bcSignalStep} of ${bcSignalSteps}`;
+
+            document.getElementById(`step-${bcSignalStep}`).classList.add('active');
+            if (prevStep) {
+                document.getElementById(`step-${prevStep}`).classList.remove('active');
+            }
+        }
+    }
+
+    /**
+     * Resets all the controls to initial state
+     */
+    function resetControls () {
+        for (const browser of document.querySelectorAll('.browser.selected')) {
+            browser.classList.remove('selected');
+        }
```

Comment:
Apologies for this, but here also stick to using `var` as well as a plain old `for` loop. We are not running this through Babel yet, and so we need to be careful about which porion of ES6+ we use.